agent/lib/broker_controller.js

```diff
@@ -476,18 +541,50 @@ BrokerController.prototype.create_address = function (a) {
 BrokerController.prototype.create_address_and_settings = function (a, settings) {
     var self = this;
     log.info('[%s] Creating address-settings for "%s": %j...', self.id, a.address, settings);
-    return self.broker.addAddressSettings(a.address, settings).then(function() {
-        log.info('[%s] Created address-settings for "%s": %j', self.id, a.address, settings);
-        return self.create_address(a);
-    }).catch(function (error) {
-        log.error('[%s] Failed to create address settings for "%s": %s', self.id, a.address, error);
-        return self.create_address(a);
-    });
+    return self.broker.getAddressSettings(a.address)
+        .then(function(oldsettings) {
+            return self.broker.addAddressSettings(a.address, settings)
+                .then(function () {
+                    log.info('[%s] Created address-settings for "%s": %j', self.id, a.address, settings);
+                    return self.broker.getAddressSettings(a.address)
+                        .then(function (newsettings) {
+                            return self.create_address(a);
+                        })
+                        .catch(function (error) {
+                            log.error('[%s] Failed to create address settings for "%s": %s', self.id, a.address, error);
+                            return self.create_address(a);
+                        });
+                });
+        });
 };
 
+BrokerController.prototype.ensure_connector = function (a) {
```

Comment:
Can you tell me more about this part of the change?

Reply:
wasn't required